src/argilla/server/contexts/datasets.py

```diff
@@ -268,14 +268,42 @@ async def list_records_by_dataset_id(
     db: "AsyncSession",
     dataset_id: UUID,
     include: List[RecordInclude] = [],
+    response_statuses: List[ResponseStatusFilter] = [],
     offset: int = 0,
     limit: int = LIST_RECORDS_LIMIT,
 ) -> List[Record]:
-    query = select(Record).filter(Record.dataset_id == dataset_id)
+    response_statuses_ = [
+        ResponseStatus(response_status)
+        for response_status in response_statuses
+        if response_status != ResponseStatusFilter.missing
+    ]
+
+    response_status_filter_expressions = []
+
+    if response_statuses_:
+        response_status_filter_expressions.append(Response.status.in_(response_statuses_))
+
+    if ResponseStatusFilter.missing in response_statuses:
+        response_status_filter_expressions.append(Response.status.is_(None))
```

Comment:
Could we move this to a function and reuse it in `list_records_by_dataset_id` and `list_records_by_dataset_id_and_user_id` functions?

Reply:
Maybe we could even combine those functions into one and use the condition `User.id == user_id` just when `user_id is not None`?

Reply:
yes, much better